In [4]:
import pandas as pd
import numpy as np
from math import *
from sklearn.ensemble import RandomForestClassifier

In [22]:
#Importing train and test data
test_df = pd.read_csv('/Users/tuhenasen/Downloads/FaceAll/FaceAll_TEST.txt',delimiter='\s+', header=None)
train_df = pd.read_csv('/Users/tuhenasen/Downloads/FaceAll/FaceAll_TRAIN.txt', delimiter='\s+',header=None)
train_df

0         1         2         3         4         5         6    \
0     1.0 -0.247592 -0.332503 -0.632105 -1.024779 -1.426342 -1.580426   
1     1.0 -0.641577 -0.939420 -1.300174 -1.385949 -1.269212 -1.036537   
2     1.0 -1.672048 -1.881515 -2.090981 -1.697734 -1.295070 -0.822547   
3     1.0 -0.463782 -1.111783 -1.747124 -2.112366 -2.409313 -2.011936   
4     1.0 -0.764739 -1.118529 -1.464805 -1.650747 -1.790880 -1.465283   
..    ...       ...       ...       ...       ...       ...       ...   
555  14.0 -0.731943 -0.927104 -0.975179 -0.880569 -0.540094  0.039170   
556  14.0 -0.735390 -1.387504 -1.678834 -1.386017 -0.531505  0.226435   
557  14.0 -0.580100 -0.915370 -1.203521 -1.196312 -0.886369 -0.516650   
558  14.0 -0.496673 -0.914445 -1.255357 -1.115175 -0.800954 -0.436291   
559  14.0 -1.374355 -1.217457 -0.655288 -0.199584 -0.006927  0.068586   

          7         8         9    ...       122       123       124  \
0   -1.336747 -1.060614 -0.723881  ... -0.582323 -0.273452 -0.077336   
1   -0.758694 -0.631150 -0.581992  ... -0.141491 -0.607773 -0.939176   
2   -0.347806  0.737430  1.852206  ... -0.298466  0.058114  0.370807   
3   -1.557494 -0.735304  0.059425  ... -0.114010 -0.061193 -0.062250   
4   -1.153654 -0.932041 -0.664873  ... -0.229956 -0.372712 -0.489895   
..        ...       ...       ...  ...       ...       ...       ...   
555  0.613808  1.000824  1.142882  ...  0.345136  0.532939  0.237443   
556  0.588829  0.812825  0.980297  ...  0.575592  0.377690  0.520676   
557 -0.226453 -0.044309  0.308878  ...  0.350632  0.819535  1.006121   
558 -0.295573 -0.050966  0.330974  ...  1.049078  1.154665  0.964952   
559  0.288190  0.561241  0.788686  ... -1.240813  1.306336  1.451640   

          125       126       127       128       129       130       131  
0   -0.119991 -0.350027 -0.490925 -0.142705  0.027710  0.090790 -0.327312  
1   -0.640406 -0.191614  0.062604  0.030992 -0.388804 -0.581837 -0.248630  
2    0.014227 -0.325895 -0.325895 -0.325895 -0.325895 -0.325895 -0.325895  
3   -0.062439 -0.062439 -0.370382 -0.726110 -0.760803 -0.763920 -0.763920  
4   -0.414688 -0.297505 -0.178539 -0.059297 -0.265746 -0.504230 -0.504230  
..        ...       ...       ...       ...       ...       ...       ...  
555 -0.104049 -0.110769  0.075533  0.244013  0.282945  0.341504  0.433592  
556  0.079867 -0.448830 -0.889983 -1.549666 -1.860682 -1.630553 -0.826028  
557  1.186400  1.332826  0.826573 -0.200211 -1.297831 -1.458397 -0.873657  
558  0.632890  0.030704 -0.348415 -0.507611 -0.575323 -1.077595 -1.595360  
559 -0.071775 -2.178860 -2.713232 -0.759848  1.881467  2.623742  0.833243  

[560 rows x 132 columns]

In [58]:
#Creating train x and y and test x and y

train_x = train_df.drop(train_df.columns[0], axis = 1)
train_y = train_df.iloc[:,0]
test_x = test_df.drop(train_df.columns[0], axis = 1)
test_y = test_df.iloc[:,0]


In [24]:
model1= RandomForestClassifier(n_estimators=10,min_samples_leaf = 1)
model1.fit(train_x,train_y)



RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=10,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [25]:
model1.score(test_x,test_y)

0.6082840236686391

In [26]:
model2= RandomForestClassifier(n_estimators=10,min_samples_leaf = 10)
model2.fit(train_x,train_y)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=10, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=10,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [27]:
model2.score(test_x,test_y)

0.6224852071005917

In [28]:
model3= RandomForestClassifier(n_estimators=10,min_samples_leaf = 20)
model3.fit(train_x,train_y)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=20, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=10,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [29]:
model3.score(test_x,test_y)

0.5763313609467455

In [30]:
model4= RandomForestClassifier(n_estimators=50,min_samples_leaf = 1)
model4.fit(train_x,train_y)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=50,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [31]:
model4.score(test_x,test_y)

0.806508875739645

In [32]:
model5= RandomForestClassifier(n_estimators=50,min_samples_leaf = 10)
model5.fit(train_x,train_y)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=10, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=50,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [33]:
model5.score(test_x,test_y)

0.7609467455621302

In [34]:
model6= RandomForestClassifier(n_estimators=50,min_samples_leaf = 20)
model6.fit(train_x,train_y)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=20, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=50,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [35]:
model6.score(test_x,test_y)

0.6171597633136094

In [36]:
model7= RandomForestClassifier(n_estimators=100,min_samples_leaf = 1)
model7.fit(train_x,train_y)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=100,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [37]:
model7.score(test_x,test_y)

0.7674556213017751

In [47]:
model8= RandomForestClassifier(n_estimators=100,min_samples_leaf = 10)
model8.fit(train_x,train_y)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=10, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=100,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [48]:
model8.score(test_x,test_y)

0.7692307692307693

In [49]:
model9= RandomForestClassifier(n_estimators=100,min_samples_leaf = 20)
model9.fit(train_x,train_y)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=20, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=100,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [50]:
model9.score(test_x,test_y)

0.7153846153846154

In [51]:
model10= RandomForestClassifier(n_estimators=500,min_samples_leaf = 1)
model10.fit(train_x,train_y)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=500,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [52]:
model10.score(test_x,test_y)

0.8272189349112427

In [54]:
model11= RandomForestClassifier(n_estimators=500,min_samples_leaf = 10)
model11.fit(train_x,train_y)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=10, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=500,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [55]:
model11.score(test_x,test_y)

0.7710059171597633

In [56]:
model12= RandomForestClassifier(n_estimators=500,min_samples_leaf = 20)
model12.fit(train_x,train_y)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=20, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=500,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [57]:
model12.score(test_x,test_y)

0.6840236686390533

OUT OF ALL THE CASES , CASE WITH N_ESTIMATORS = 500 AND MIN_SAMPLES_LEAF=1 HAS THE MOST ACCURACY AS 82%
